gateway/src/node/mod.rs

```diff
@@ -565,12 +619,18 @@ impl<St> Gateway<St> {
 
         #[cfg(feature = "wireguard")]
         let _wg_api = if let Some(opts) = self.authenticator_opts.clone() {
```

Comment:
nitpicking: you don't have to clone it, you can just use `as_ref()` since `start_authenticator` takes a reference anyway

Reply:
clippy could capture things like this, do we run it in the CI?

Reply:
The reason it needs to be like this and clippy didn't say anything is because the `opts` of `self` is used in a function that also takes `self` as mutable. But I've moved things a bit so that `opts` is unwrapped inside the function, which is a bit more natural.